### PR TITLE
Modifications To Strong match URL

### PR DESCRIPTION
**Pull request by sojanpr**
1. Making Sure GAID is always added to strong match url
2. Checking strong match in case of random HW ID
3. Random HW ID is added to Strong match  url if HWID fetch is disabled
4. Adding HW ID type (`vendor_id`, `random`)  to strong match url

Modified strong match url look like this
https://app.link/_strong_match?os=Android&hardware_id=8010918c-e28e-40cc-8961-333898e6c520&hardware_id_type=random&google_advertising_id=1da873b4-71a9-4f13-a0f9-XXXXXXXX&app_version=2.5.7&branch_key=key_test_XXXXXX&sdk=android2.5.7

@aaustin @dmitrig01 @EvangelosG

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java [hunk 1]
```diff
@@ -144,6 +144,8 @@ private Uri buildStrongMatchUrl(String matchDomain, DeviceInfo deviceInfo, PrefH
             String uriString = "https://" + matchDomain + "/_strong_match?os=" + deviceInfo.getOsName();
             // Add HW ID
             uriString += "&" + Defines.Jsonkey.HardwareID.getKey() + "=" + deviceInfo.getHardwareID();
+            String hardwareIDTypeVal = deviceInfo.isHardwareIDReal() ? Defines.Jsonkey.HardwareIDTypeVendor.getKey() : Defines.Jsonkey.HardwareIDTypeRandom.getKey();
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java]: Won't `HardwareIDTypeRandom` return a different random variable here than when used for `v1/install`?

Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java]: the `hardware_id` must match between the two for strong matching to work

Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java]: `hardware_id ` is read from DeviceInfo class. DeviceInfo is a singleton class and `hardware_id ` variable is initialized once in a session

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 2]
```diff
@@ -394,13 +392,16 @@
...
-    private static final String[] EXTERNAL_INTENT_EXTRA_KEY_WHITE_LIST = new String[] {
+    private static final String[] EXTERNAL_INTENT_EXTRA_KEY_WHITE_LIST = new String[]{
             "extra_launch_uri"   // Key for embedded uri in FB ads triggered intents
     };
 
     private CountDownLatch getFirstReferringParamsLatch = null;
     private CountDownLatch getLatestReferringParamsLatch = null;
 
+    /* Flag for checking of Strong matching is waiting on GAID fetch */
+    private boolean performCookieBaseDStrongMatchingOnGAIDAvailable = false;
```
Comment (EvangelosG) [inline Branch-SDK/src/io/branch/referral/Branch.java]: performCookieBaseDStrongMatchingOnGAIDAvailable - is the "D" upper case for a reason or typo?

Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/Branch.java]: Typo. Corrected

**Comment by EvangelosG** [comment]
My tests show HWID in strong match URL and v1/install post are equal. 👍

**Pull request merged**